freqtrade/exchange/exchange.py

```diff
@@ -877,42 +877,67 @@ def get_dry_market_fill_price(self, pair: str, side: str, amount: float, rate: f
 
         return rate
 
-    def _is_dry_limit_order_filled(self, pair: str, side: str, limit: float) -> bool:
+    def _fill_dry_limit_order(self, pair: str, side: str,
+                              limit: float, given_amount: float) -> Tuple[Optional[float], float]:
+        """
+        Returns average price and filled amount
+        """
         if not self.exchange_has('fetchL2OrderBook'):
-            return True
-        ob = self.fetch_l2_order_book(pair, 1)
-        try:
+            return limit, given_amount
+        for order_book_top in (None, 1000):
+            ob = self.fetch_l2_order_book(pair, order_book_top)
```

Comment:
Passing `None` as value here is not supported and leads to an exception - and i'm actually surprised mypy isn't screaming at you - as tye type-hint on `fetch_l2_order_book()` defines it as int - not `Optional[int]`.

I'm anyway not sure why you're using None as first argument (or why you're looping at all). 
If you'd like to fetch one level only, i'd use `1` instead of None to get just one level (i'm not sure why though - as that might lead to 2 requests which are not really necessary).

Reply:
Coz mypy is only checking for freqtrade and scripts folder.

Reply:
mypy is checking all files within the freqtrade directory (including this one).
I'm therefore not sure what you mean with your comment.

It's failing to properly detect this - which is probably a bug in mypy (caused by the decorator it appears - as removing the `@retrier` decorator causes mypy to point out errors).

Even if mypy is failing to detect this (this can also happen in other cases, where the type on the function is missing for example) - that's no excuse for submitting untested PR's.